morango/models.py

```diff
@@ -305,21 +306,21 @@ def _deserialize_store_model(self):
                     pass
             else:
                 klass_model.objects.filter(id=self.id).delete()
-            return valid  # mark deletion as being validated
+            return None, valid  # mark deletion as being validated
         else:
             # load model into memory
             app_model = klass_model.deserialize(json.loads(self.serialized))
             app_model._morango_source_id = self.source_id
             app_model._morango_partition = self.partition
+            app_model._morango_dirty_bit = False
 
             try:
-                # validate and save the model
+                # validate the model
                 app_model.clean_fields()
-                with mute_signals(signals.pre_save, signals.post_save):
-                    app_model.save(update_dirty_bit_to=False)
-                return valid
-            except exceptions.ValidationError:
+                return app_model, valid
+            except exceptions.ValidationError as e:
                 valid = False
+                logger.debug("Validation error for {model} with id {id}: {error}".format(model=klass_model.__name__, id=app_model.id, error=e))
```

Comment:
Might be good to at least have this as a `logger.warn`, so it will show up in production logs people might share with us.